File: app/src/main/java/com/example/android/toastapp/MainActivity.java
package com.example.android.toastapp;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {
    private int clickCount=0;
    private TextView showClickCount;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        showClickCount= (TextView) findViewById(R.id.show_count);
    }

    public void showToast(View view) {
        Toast toast=Toast.makeText(this, R.string.toast_message, Toast.LENGTH_SHORT);
        toast.show();
    }

    public void countUp(View view) {
        clickCount++;
        if (showClickCount!=null){
            showClickCount.setText(Integer.toString(clickCount));
        }
    }
}

File: app/src/main/java/com/example/android/toastapp/MainActivity.kt
package com.example.android.toastapp

import androidx.appcompat.app.AppCompatActivity

import android.os.Bundle
import android.os.PersistableBundle
import android.view.View
import android.widget.Button
import android.widget.TextView
import android.widget.Toast

class MainActivity : AppCompatActivity() {
    private var clickCount = 0
    private var showClickCount: TextView? = null
    private var clickButton:Button?=null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        showClickCount = findViewById<View>(R.id.show_count) as TextView
        clickButton = findViewById<View>(R.id.button_count) as Button

        if (savedInstanceState!=null){
            clickCount=savedInstanceState.getInt("COUNT")
            showClickCount!!.text = Integer.toString(clickCount)

        }

        clickButton?.setOnLongClickListener(View.OnLongClickListener {
             v:View->
                clickCount=0;
                showClickCount?.setText(Integer.toString(clickCount))
                true
        })
    }

    fun showToast(view: View) {
        val toast = Toast.makeText(this, R.string.toast_message, Toast.LENGTH_SHORT)
        toast.show()
    }

    fun countUp(view: View) {
        clickCount++
        if (showClickCount != null) {
            showClickCount!!.text = Integer.toString(clickCount)
        }
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        outState.putInt("COUNT", clickCount);
    }
}
